presto-main/src/main/java/com/facebook/presto/sql/planner/Plan.java

```diff
@@ -40,4 +71,38 @@ public TypeProvider getTypes()
     {
         return types;
     }
+
+    public Map<PlanNodeId, PlanNodeStatsEstimate> getStats()
+    {
+        return stats;
+    }
+
+    public Map<PlanNodeId, PlanNodeCostEstimate> getCosts()
+    {
+        return costs;
```

Comment:
I don;t quite like this being part of the plan -- would it be possible to keep it separate and pass further down separately?

Reply:
Yeah, we can have a separate class `StatsAndCosts` that contains both `stats` and `costs` maps, and we can pass it all around as a single object. I think it will be even nicer.